src/main/java/com/example/MySmartHomeApp.java

```diff
@@ -70,25 +70,30 @@ public SyncResponse onSync(SyncRequest syncRequest, Map<?, ?> headers) {
     res.payload.devices = new SyncResponse.Payload.Device[numOfDevices];
     for (int i = 0; i < numOfDevices; i++) {
       QueryDocumentSnapshot device = devices.get(i);
+      HashMap name = new HashMap();
+      name = (HashMap)device.get("name");
+      HashMap deviceInfo = new HashMap();
+      deviceInfo = (HashMap)device.get("deviceInfo");
+
       SyncResponse.Payload.Device.Builder deviceBuilder =
           new SyncResponse.Payload.Device.Builder()
               .setId(device.getId())
               .setType((String) device.get("type"))
               .setTraits((List<String>) device.get("traits"))
               .setName(
                   DeviceProto.DeviceNames.newBuilder()
-                      .addAllDefaultNames((List<String>) device.get("defaultNames"))
-                      .setName((String) device.get("name"))
-                      .addAllNicknames((List<String>) device.get("nicknames"))
+                      .addAllDefaultNames((List<String>) name.get("defaultNames"))
```

Comment:
In the expected data model, coming from the frontend, the [name](https://github.com/actions-on-google/smart-home-frontend/blob/master/src/device-types/ac-unit.js#L490) is a top-level field

Reply:
Thanks, I got it!